src/julia_internal.h

```diff
@@ -231,7 +231,7 @@ JL_DLLEXPORT extern const char *jl_filename;
 
 jl_value_t *jl_gc_pool_alloc_noinline(jl_ptls_t ptls, int pool_offset,
                                    int osize);
-JL_DLLEXPORT jl_value_t *jl_gc_big_alloc(jl_ptls_t ptls, size_t allocsz);
```

Comment:
This seems wrong? You want to dllexport `jl_gc_big_alloc`

Reply:
Eep, i think you're mostly right! Gosh sorry this is so tricky. C declarations...

You're right: i want to dllexport `jl_gc_big_alloc`. But I _am_ already doing that, because the _definition_ is marked `dllexport`.

You're also right that I _shouldn't_ be dllexporting `jl_gc_big_alloc_noinline`. I'll remove that now.

But _I'm right_ (I'm pretty sure) that we want to delete jl_gc_big_alloc from the header file, since it's not meant to be called directly from the C code anywhere, so it doesn't need to be forward declared. It's simply emitted by the LLVM codegen, and since it's dllexported in the _object file_, it gets linked correctly! :)

So i think the only change to make here is to drop the `JL_DLLEXPORT` from jl_gc_big_alloc_noinline.

I'll do that now.

-----

This is the pattern we followed in the previous PR:
https://github.com/JuliaLang/julia/pull/43868/files#diff-7b3edb5362e7b70b75aff15c04b492aa3c74b883fca8eeee74d7bd839f81e4b0L231-R232

Reply:
I've made this change. thanks!